lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(BadConfig) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert config.logger_host == "logger.example.com"
+            assert exc == "No option 'logger_port' in section: 'logging'"
+
+    def test_logger_type_port_value(self, monkeypatch):
+        """verify `port` value of logger_type `hostport`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(ValueError) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert config.logger_host == "logger.example.com"
+            assert exc == "No option 'logger_port' in section: 'logging'"
+
+    def test_logger_hostport(self, monkeypatch):
+        """verify logger_type as `hostport` with `host` and `port`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "42",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "hostport"
+        assert config.logger_host == "logger.example.com"
+        assert config.logger_port == "42"
+
+    def test_no_logger_dir_option(self, monkeypatch):
+        """test no logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_dir is None
```

Comment:
This test looks to me to be identical to `test_logger()` except for the assertion at line 168.  I recommend that you move the assertion from here to that test and omit this test.

Reply:
sure